pool/src/lib.rs

```diff
@@ -303,30 +350,56 @@ impl LendingPoolTrait for LendingPool {
     /// - amount - sended amount of s-token
     /// - balance_from_before - amount of s-token before transfer on `from` user balance
     /// - balance_to_before - amount of s-token before transfer on `to` user balance
+    /// # Panics
+    ///
+    /// Panics if the caller is not the sToken contract.
     fn finalize_transfer(
         env: Env,
         asset: Address,
         from: Address,
-        _to: Address,
+        to: Address,
         amount: i128,
         balance_from_before: i128,
-        _balance_to_before: i128,
+        balance_to_before: i128,
     ) -> Result<(), Error> {
-        let reserve = get_actual_reserve_data(&env, asset)?;
+        // TODO: maybe check with callstack?
+        let reserve = get_actual_reserve_data(&env, asset.clone())?;
         reserve.s_token_address.require_auth();
         Self::require_not_paused(&env)?;
-        // TODO
         let balance_from_after = balance_from_before
             .checked_sub(amount)
             .ok_or(Error::InvalidAmount)?;
 
-        Self::require_good_position(
+        let mut from_config = read_user_config(&env, from.clone())?;
+        let reserves = read_reserves(&env);
+        let account_data = Self::calc_account_data(
             &env,
-            from,
-            Some((reserve.s_token_address, balance_from_after)),
-            None,
-            true,
+            from.clone(),
+            Some(AssetBalance::new(
+                reserve.s_token_address,
+                balance_from_after,
+            )),
+            &from_config,
+            &reserves,
+            false,
         )?;
+        Self::require_good_position(account_data)?;
+
+        if from != to {
+            let reserve_id = read_reserve(&env, asset.clone())?.get_id();
```

Comment:
I see that you've already read the asset reserve at the line 366. Can we use it to get reserve it here?